tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/factoriestransformers/SaslFactoriesTest.java

```diff
@@ -0,0 +1,628 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.hal.testsuite.test.configuration.elytron.factoriestransformers;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.hal.resources.Names;
+import org.jboss.hal.testsuite.fragment.AddResourceDialogFragment;
+import org.jboss.hal.testsuite.fragment.FormFragment;
+import org.jboss.hal.testsuite.fragment.TableFragment;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.jboss.arquillian.graphene.Graphene.waitGui;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.FILTERS;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.MECHANISM_CONFIGURATIONS;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.MECHANISM_NAME;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.MECHANISM_REALM_CONFIGURATIONS;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.MODULE;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.NAME;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.PATTERN_FILTER;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.PROTOCOL;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.PROVIDER_NAME;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.REALM_NAME;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.SECURITY_DOMAIN;
+import static org.jboss.hal.resources.Ids.ELYTRON_CONFIGURABLE_SASL_SERVER_FACTORY;
+import static org.jboss.hal.resources.Ids.ELYTRON_MECHANISM_PROVIDER_FILTERING_SASL_SERVER_FACTORY;
+import static org.jboss.hal.resources.Ids.ELYTRON_SASL_AUTHENTICATION_FACTORY;
+import static org.jboss.hal.testsuite.test.configuration.elytron.ElytronFixtures.*;
+
+@RunWith(Arquillian.class)
+public class SaslFactoriesTest extends AbstractFactoriesTransformersTest {
+
```

Comment:
Similarly to the principal transformers, I would split this test class into the corresponding factories test classes e.g AggregateSaslServerFactoryTest, ConfigurableSaslServerFactoryTest etc.